ansible/roles/marathon-server/tasks/main.yml

```diff
@@ -173,16 +173,18 @@
       - marathon
       - plugin
       - permissions
+      - permissions-sync
 
   - name: Configure Marathon http_address commandline flag
     template: src=marathon-flags/http_address.j2 dest=/etc/marathon/conf/http_address owner=root group=root mode=0644
-    when: nginx_auth_enable is defined and nginx_auth_enable|bool and (marathon_auth_enable is not defined or not marathon_auth_enable|bool)
+    when: nginx_auth_enable is defined and nginx_auth_enable|bool and marathon_auth_enable is defined and marathon_auth_enable|bool
```

Comment:
I don't think the above line should change.
The new code does not take care of the situation when `marathon_auth_enable` is not defined (and only `nginx_auth_enable` is defined.
Actually, do we even need this `when` condition? This has nothing to do with auth, but more to do with whether dns/proxy is set up. Right?